kong/identifiers.go

```diff
@@ -0,0 +1,91 @@
+package kong
+
+import "fmt"
+
+// Identifier returns the endpoint key name or ID.
+func (s1 *Service) Identifier() string {
+	if s1.Name != nil {
+		return *s1.Name
+	}
+	return *s1.ID
```

Comment:
For all types please only dereference ID if it's non-nil.

If none of the checked fields is non-nil, return `""` as the friendly name.